Library/Homebrew/utils.rb

```diff
@@ -291,6 +291,16 @@ def paths
   end.uniq.compact
 end
 
+# return the shell profile file based on users' preference shell
+def shell_profile
+  case ENV["SHELL"]
+  when %r{/(ba)?sh} then "~/.bash_profile"
+  when %r{/zsh} then "~/.zshrc"
+  when %r{/ksh} then "~/.kshrc"
+  else "~/.profile"
```

Comment:
There're considerable among people(include myself) using `zsh`. And I do see the issues related to it now and then.